env.nix

```diff
@@ -0,0 +1,315 @@
+        });
+
+        doctest = overrideCabal super.doctest (drv: {
+          version = "0.9.13";
+          revision = "1";
+          sha256 = "0xl570ay5bw1rpd1aw59c092rnwjbp9qykh2rhpxyvl333p8mg00";
+          editedCabalFile = "592ab6d62eca8a0b43930f15c8fb653c54d60983bd232ecc505bd5a5aebe6f7f";
+        });
+
+        haskell-src-meta = overrideCabal super.haskell-src-meta (drv: {
+          version = "0.6.0.10";
+          sha256 = "0flcyimibz4flq66isshn2zsmzlly6sja6gfb0a0xn4ns4xpwpy1";
+        });
+
+        haddock = overrideCabal super.haddock (drv: {
+          version = "2.16.1";
+          sha256 = "1mnnvc5jqp6n6rj7xw8wdm0z2xp9fndkz11c8p3vbljsrcqd3v26";
+          doCheck = false;
+        });
+
+        haddock-api = overrideCabal super.haddock-api (drv: {
+          version = "2.16.1";
+          sha256 = "1spd5axg1pdjv4dkdb5gcwjsc8gg37qi4mr2k2db6ayywdkis1p2";
+          doCheck = false;
+        });
+
+        haddock-library = overrideCabal super.haddock-library (drv: {
+          version = "1.2.1";
+          sha256 = "0mhh2ppfhrvvi9485ipwbkv2fbgj35jvz3la02y3jlvg5ffs1c8g";
+          doCheck = false;
+        });
+
+        QuickCheck = overrideCabal super.QuickCheck (drv: {
+          version = "2.8.1";
+          sha256 = "0fvnfl30fxmj5q920l13641ar896d53z0z6z66m7c1366lvalwvh";
+        });
+
+        exceptions = overrideCabal super.exceptions (drv: {
+          jailbreak = true;
+        });
+
+        cassava = overrideCabal super.cassava (drv: {
+          jailbreak = true;
+        });
+
+        gps = dontCheck super.gps; # The test suite for gps pulls in gpx-conduit, which doesn't build
+      };
+    };
+in rec {
+  inherit overrideCabal extendHaskellPackages;
+
+  ghc = extendHaskellPackages pkgs.haskell.packages.ghc7102;
+
+  ghcjsCompiler = overrideCabal (ghc.callPackage "${path}/pkgs/development/compilers/ghcjs" {
+    ghc = pkgs.haskell.compiler.ghc7102;
+
+    ghcjsBoot = fetchgit {
+      url = git://github.com/ghcjs/ghcjs-boot.git;
+      rev = "d435c60b62d24b7a4117493f7aaecbfa09968fe6"; # 7.10 branch
+      sha256 = "4159b20730822ec699b47036791158bc32d5412903005d19c396b120beac701f";
+      fetchSubmodules = true;
+    };
+
+    shims = fetchgit {
+      url = git://github.com/ghcjs/shims.git;
+      rev = "0b670ca27fff3f0bad515c37e56ccb8b4d6758fb"; # master branch
+      sha256 = "08c0c3547e06d7716b2feb6ebda02f2ab33c205700848ad8e134152f5c3af8a7";
+    };
+  }) (drv: {
+    src = fetchgit {
+      url = git://github.com/ghcjs/ghcjs.git;
+      rev = "fb1faa9cb0a11a8b27b0033dfdb07aafb6add35e"; # master branch
+      sha256 = "9069f484da55bf5af8dc65e539f86ca5e1b64ab9ecef65f38006c954400a0eef";
+    };
+  });
+
+  ghcjsPackages = callPackage "${path}/pkgs/development/haskell-modules" {
+    ghc = ghcjsCompiler;
+    packageSetConfig = callPackage "${path}/pkgs/development/haskell-modules/configuration-ghcjs.nix" { };
+  };
+
+  ghcjs = extendHaskellPackages ghcjsPackages;
+}
```

Comment:
Since we're taking out the makeRecursivelyOverridable, we should add a test case that verifies that re-overriding does in fact work.